lib/wings/active_fedora_converter.rb

```diff
@@ -82,34 +79,34 @@ def active_fedora_class
     # then the id hasn't been minted and shouldn't yet be set.
     # @return [String]
     def id
-      id_attr = resource[:id]
-      return id_attr.to_s if id_attr.present? && id_attr.is_a?(::Valkyrie::ID) && !id_attr.blank?
+      return resource[:id].to_s if resource[:id]&.is_a?(::Valkyrie::ID) && !resource[:id].blank?
       return "" unless resource.respond_to?(:alternate_ids)
+
       resource.alternate_ids.first.to_s
     end
 
     def self.DefaultWork(resource_class)
       Class.new(DefaultWork) do
         self.valkyrie_class = resource_class
 
-        # nested attributes in AF don't inherit! this needs to be here until
-        # we can drop it completely.
+        resource_class.schema.each do |schema_key|
+          next if resource_class.reserved_attributes.include?(schema_key.name) ||
+                  protected_property_name?(schema_key.name) ||
+                  properties.keys.include?(schema_key.name.to_s)
+          property schema_key.name, predicate: RDF::URI("http://hyrax.samvera.org/ns/wings##{schema_key.name}")
```

Comment:
@cjcolvar you had a question in Samvera tech, an you write that up here?

Reply:
the question was, briefly: can we define the predicate URIs through valkyrie configuration (and ultimately through Hyrax's YAML driven metadata configuration?

---

yes, i think we should explore this.

can we consider rolling forward with this and taking that work on as immediate follow-up? are there SemVer concerns about that approach?

Reply:
I think this could be done as a follow up if we allow customizing predicates falling back to these default predicates if none customized.

I think we could follow an approach similar to that taken in the fedora adapter in valkyrie.
The metadata adapter there can be initialized with a schema (https://github.com/samvera/valkyrie/blob/master/lib/valkyrie/persistence/fedora/metadata_adapter.rb#L9) which responds to `#predicate_for` and `#property_for`(https://github.com/samvera/valkyrie/blob/master/lib/valkyrie/persistence/fedora/permissive_schema.rb#L74-L97) and used in the ModelConverter and OrmConverter.